ngraph_bridge/ngraph_executor.cc

```diff
@@ -240,190 +220,167 @@ Status NGraphExecutor::GetNgExecutable(
+        serialized_ng_func);
+    if (status != Status::OK()) {
+      return std::make_pair(status, std::make_pair(ng_exec, ng_function));
     }
-    ng_exec = it->second;
-    cache_hit = true;
-    NGRAPH_VLOG(1) << "Compilation cache hit: " << m_node_name;
+#endif
   }
-  return Status::OK();
+  // Get NgExecutable
+  auto status_ng_exec_pair =
+      GetNgExecutable(signature, ng_function, serialized_ng_func, op_backend);
+  // Create PipelinedTensorStore
+  if (status_ng_exec_pair.first == Status::OK()) {
+    ng_exec = status_ng_exec_pair.second;
+    auto status_ng_pts_pair = InitializeIOTensorPipeline(ng_exec);
+    pts = status_ng_pts_pair.second;
+    return std::make_pair(status_ng_pts_pair.first,
+                          std::make_tuple(ng_exec, serialized_ng_func, pts));
+  }
+  return std::make_pair(status_ng_exec_pair.first,
+                        std::make_tuple(ng_exec, serialized_ng_func, pts));
 }
 
 //---------------------------------------------------------------------------
-//  GetNgFunction
+//  NGraphExecutor::GetNgExecutable
 //---------------------------------------------------------------------------
-Status NGraphExecutor::GetNgFunction(
-    const std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
-    std::shared_ptr<ngraph::Function>& ng_function) {
-  // Lookup the function from the Map
-  auto it = m_serialized_ng_function_map.find(ng_exec);
-  if (it == m_serialized_ng_function_map.end()) {
-    return errors::Internal("Function not found for this executable");
-  }
+std::pair<Status, std::shared_ptr<ngraph::runtime::Executable>>
+NGraphExecutor::GetNgExecutable(std::string signature,
+                                std::shared_ptr<ngraph::Function>& ng_function,
+                                const string serialized_ng_func,
+                                ng::runtime::Backend*& op_backend) {
+  std::shared_ptr<ngraph::runtime::Executable> ng_exec;
+
+  ngraph::Event event_compile("Compile nGraph", m_node_name, "");
+  BackendManager::LockBackend(m_op_backend_name);
   try {
-    ng_function = ngraph::deserialize(it->second);
+    if (m_do_aot) {
+      auto itr = m_aot_execs.find(signature);
+      if (itr == m_aot_execs.end()) {
+        BackendManager::UnlockBackend(m_op_backend_name);
+        return std::make_pair(
+            errors::Internal(
+                "Requested AOT, but could not find string with the "
+                "signature: ",
+                signature),
+            nullptr);
+      }
+      stringstream serialized_exec_read;
+      serialized_exec_read << (itr->second);
+      ng_exec = op_backend->load(serialized_exec_read);
+    } else {
+      ng_exec = op_backend->compile(ng_function);
+    }
   } catch (const std::exception& exp) {
-    return errors::Internal("Failed to deserialize ngraph function");
+    BackendManager::UnlockBackend(m_op_backend_name);
+    Status st = StringToFile("tf_function_error_" + m_node_name + ".json",
```

Comment:
we can move this dumping of serialized ngfunction in the calling function. we dont need to send both ng-func and serialized ng-func in that case

Reply:
Idea is too keep  executable creation logic separate from calling, I think what you are suggesting would defeat the purpose. If you want we make discuss it.

Reply:
Please change as discussed.